src/main/scala/com/microsoft/azure/cosmosdb/spark/rdd/CosmosDBRDDIterator.scala

```diff
@@ -367,7 +368,11 @@ class CosmosDBRDDIterator(hadoopConfig: mutable.Map[String, String],
           case ex: IllegalStateException =>
             if (ex.getCause.isInstanceOf[ServiceUnavailableException]) {
               if (retryCount < maxRetryCountOnServiceUnavailable) {
-                logWarning(s"Service Unavailable exception thrown. Going to retry. Current retry count: ${retryCount}. Max retry count: ${maxRetryCountOnServiceUnavailable}")
+                val retryDelayInMs = rnd.nextInt(1000)
```

Comment:
Why a random value here? we usually recommend an exponential increase for a fix number of retries... Like start with 100 ms and multiply that by a factor of the retry count.

Reply:
Partition split triggers this code path roughly at the same time on all partitions. Using randomization to avoid that retries happen in-sync on all partitions

Reply:
we probably should use exponential retry instead of random.